### ISSUE TITLE
Index out of range error

### ISSUE THREAD

**Issue by eyomi-gilead** [python-semantic-release/python-semantic-release#669]
## The problem

when I run `semantic-release publish` on my project for the first time it fails with an `Index out of range` error. 

## Expected behavior

`Index out of range` is not a very useful error as it gives no clue as to the underlying issue. It is more of symptom than a cause. It would be better to detect and prevent it, and display a more suitable user-friendly error that can help troubleshoot the issue. I have been stuck on this issue for 2 days now, and still I have no clue what I'm doing wrong. 2 projects seemingly configured identically yet one fails with this error.

## Environment
I am getting this issue when running locally (Mac OS Ventura 13.5) and in my CI/CD pipeline on Ubuntu latest.

```shell
Python 3.11.4
pip 23.2.1
semantic-release, version 8.0.4

pip freeze
black==23.7.0
boto3==1.28.19
botocore==1.31.19
certifi==2023.7.22
cffi==1.[15](https://github.com/GILEAD-DNA-PLATFORM/pdm-databricks-client-test/actions/runs/5763720306/job/15626141652#step:21:16).1
cfgv==3.3.1
charset-normalizer==3.2.0
click==8.1.6
coverage==7.2.7
cryptography==41.0.3
delta-spark==2.4.0
distlib==0.3.7
dotty-dict==1.3.1
filelock==3.12.2
flake8==6.1.0
flake8-black==0.3.6
gitdb==4.0.10
GitPython==3.1.32
identify==2.5.26
idna==3.4
importlib-metadata==6.8.0
importlib-resources==5.7.0
Jinja2==3.1.2
jmespath==1.0.1
markdown-it-py==3.0.0
MarkupSafe==2.1.3
mccabe==0.7.0
mdurl==0.1.2
-e git+https://github.com/GILEAD-DNA-PLATFORM/pdm-databricks-client-test@7c012c653f1fde05cd13a7b04ef25541966046e4#egg=metrics
moto==4.1.14
mypy-extensions==1.0.0
nodeenv==1.8.0
packaging==23.1
pathspec==0.11.2
pdoc==14.0.0
platformdirs==3.10.0
pre-commit==3.3.3
py-partiql-parser==0.3.6
py4j==0.10.9.7
pycodestyle==2.11.0
pycparser==2.21
pydantic==1.10.12
pyflakes==3.1.0
Pygments==2.15.1
pyspark==3.4.0
python-dateutil==2.8.2
python-gitlab==3.15.0
python-lorem==1.3.0.post1
python-semantic-release==8.0.4
pytz==2023.3
PyYAML==6.0.1
requests==2.31.0
requests-toolbelt==1.0.0
responses==0.23.3
rich==13.5.2
s3transfer==0.6.1
shellingham==1.5.0.post1
six==1.[16](https://github.com/GILEAD-DNA-PLATFORM/pdm-databricks-client-test/actions/runs/5763720306/job/15626141652#step:21:17).0
smmap==5.0.0
tomlkit==0.12.1
types-PyYAML==6.0.12.11
typing_extensions==4.7.1
urllib3==1.26.16
virtualenv==[20](https://github.com/GILEAD-DNA-PLATFORM/pdm-databricks-client-test/actions/runs/5763720306/job/15626141652#step:21:21).[24](https://github.com/GILEAD-DNA-PLATFORM/pdm-databricks-client-test/actions/runs/5763720306/job/15626141652#step:21:25).2
Werkzeug==2.3.6
xmltodict==0.13.0
zipp==3.16.2
```

I am using pip as the build tool

## Configuration
[tool.semantic_release]
version_variable = "src/semantic_release/__init__.py:__version__"
branch = "main"
upload_to_repository = false
upload_to_release = true
build_command = "mkdir dist && zip -r dist/coverage.zip ./htmlcov && zip -r dist/documentation.zip ./docs && pip wheel -w dist ."

## Logs

[09:56:08] DEBUG    [semantic_release.cli.commands.main] DEBUG main.main: logging level set to: DEBUG                                                                                                   main.py:105
           DEBUG    [semantic_release.cli.commands.main] DEBUG main.main: global cli options: GlobalCommandLineOptions(noop=False, verbosity=2, config_file='pyproject.toml', strict=False)             main.py:123
           INFO     [semantic_release.cli.util] INFO util.load_raw_config_file: Loading configuration from pyproject.toml                                                                                util.py:84
           DEBUG    [semantic_release.cli.util] DEBUG util.load_raw_config_file: Trying to parse configuration pyproject.toml in TOML format                                                             util.py:87
           INFO     [semantic_release.cli.config] INFO config.select_branch_options: Using group 'main' options, as '(main|master)' matches 'main'                                                    config.py:228
           WARNING  [semantic_release.cli.config] WARNING config.from_raw_config: the token for the remote VCS is configured as stored in the GH_TOKEN environment variable, but it is empty          config.py:331
           DEBUG    [semantic_release.changelog.template] DEBUG template.environment: {'template_dir': 'templates', 'block_start_string': '{%', 'block_end_string': '%}', 'variable_start_string':   template.py:51
                    '{{', 'variable_end_string': '}}', 'comment_start_string': '{#', 'comment_end_string': '#}', 'line_statement_prefix': None, 'line_comment_prefix': None, 'trim_blocks': False,                 
                    'lstrip_blocks': False, 'newline_sequence': '\n', 'keep_trailing_newline': False, 'extensions': (), 'autoescape': True, 'autoescape_value': True}                                              
           DEBUG    [semantic_release.version.translator] DEBUG translator._invert_tag_format_to_re: inverted tag_format 'v{version}' to 'v(?P<version>.*)'                                        translator.py:40
           INFO     [semantic_release.version.algorithm] INFO algorithm.tags_and_versions: found 0 previous tags                                                                                    algorithm.py:56
Traceback (most recent call last):
  File ".../bin/semantic-release", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File ".../click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File ".../click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../semantic_release/cli/commands/publish.py", line 35, in publish
    tag = str(tags_and_versions(repo.tags, translator)[0][0])
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^

```shell
           DEBUG    [semantic_release.cli.commands.main] DEBUG       main.py:105
                    main.main: logging level set to: DEBUG                      
           DEBUG    [semantic_release.cli.commands.main] DEBUG       main.py:123
                    main.main: global cli options:                              
                    GlobalCommandLineOptions(noop=False,                        
                    verbosity=2, config_file='pyproject.toml',                  
                    strict=False)                                               
           INFO     [semantic_release.cli.util] INFO                  util.py:84
                    util.load_raw_config_file: Loading configuration            
                    from pyproject.toml                                         
           DEBUG    [semantic_release.cli.util] DEBUG                 util.py:87
                    util.load_raw_config_file: Trying to parse                  
                    configuration pyproject.toml in TOML format                 
           INFO     [semantic_release.cli.config] INFO             config.py:228
                    config.select_branch_options: Using group                   
                    'main' options, as '(main|master)' matches                  
                    'main'                                                      
           DEBUG    [semantic_release.hvcs.util] DEBUG                util.py:45
                    util.build_requests_session: setting up default             
                    session authentication                                      
           DEBUG    [semantic_release.changelog.template] DEBUG   template.py:51
                    template.environment: {'template_dir':                      
                    'templates', 'block_start_string': '{%',                    
                    'block_end_string': '%}',                                   
                    'variable_start_string': '{{',                              
                    'variable_end_string': '}}',                                
                    'comment_start_string': '{#',                               
                    'comment_end_string': '#}',                                 
                    'line_statement_prefix': None,                              
                    'line_comment_prefix': None, 'trim_blocks':                 
                    False, 'lstrip_blocks': False,                              
                    'newline_sequence': '\n',                                   
                    'keep_trailing_newline': False, 'extensions':               
                    (), 'autoescape': True, 'autoescape_value':                 
                    True}                                                       
           DEBUG    [semantic_release.version.translator] DEBUG translator.py:40
                    translator._invert_tag_format_to_re:                        
                    inverted tag_format 'v{version}' to                         
                    'v(?P<version>.*)'                                          
           DEBUG    [semantic_release.cli.masking_filter]   masking_filter.py:30
                    DEBUG masking_filter.add_mask_for:                          
                    Adding redact pattern                                       
                    'context.hvcs_client.token' to                              
                    _redact_patterns                                            
           DEBUG    [semantic_release.cli.masking_filter]   masking_filter.py:30
                    DEBUG masking_filter.add_mask_for:                          
                    Adding redact pattern                                       
                    'context.hvcs_client.token' to                              
                    _redact_patterns                                            
           INFO     [semantic_release.version.algorithm] INFO    algorithm.py:56
                    algorithm.tags_and_versions: found 0                        
                    previous tags                                               
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.4/x64/bin/semantic-release", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/click/core.py", line 1[15] 7, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/click/core.py", line [16] 88, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/click/decorators.py", line [33], in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/semantic_release/cli/commands/publish.py", line [35] in publish
    tag = str(tags_and_versions(repo.tags, translator)[0][0])
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
Error: Process completed with exit code 1.
```

If using GitHub actions, ensure the `root_options` contains the `-vv` flag
for verbosity, and attach your workflow definition.

## Additional context

_Add any other information that could be useful, such as a link to your project
(if public), links to a failing GitHub Action, or an example commit._

**Comment by bernardcooke53** [comment]
Hey @eyomi-gilead :wave: 

Is this a new project? It looks like PSR isn't finding any tags for previous versions. Can you point it directly at a previous release (`semantic-release publish --tag $TAG`)?

**Comment by eyomi-gilead** [comment]
@bernardcooke53 apologies for the late reply. I was out of office for a few days. I tried the command you suggested above (semantic-release publish --tag $TAG). The command ran successfully but nothing was released and the repo was not tagged. Am I missing something?

**Comment by eyomi-gilead** [comment]
@bernardcooke53 Yes it is a new project. Is there something I am supposed to do first?

**Comment by eyomi-gilead** [comment]
The other thing I am finding out is that the version variable in ./semantic_release/__init__.py is not getting updated in this project. Also just running `semantic-release publish` does nothing

**Comment by eyomi-gilead** [comment]
@bernardcooke53 I am going around in circle with this. Here is a summary of the issues I have faced using semantic-release in my new project:
1. semantic-release publish on the new project failed with following log
```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.4/x64/bin/semantic-release", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/click/core.py", line 1[15](https://github.com/.../actions/runs/5754905253/job/15601195357#step:17:16)7, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/click/core.py", line [16](https://github.com/.../actions/runs/5754905253/job/15601195357#step:17:17)88, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/click/decorators.py", line [33](https://github.com/.../actions/runs/5754905253/job/15601195357#step:17:34), in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/semantic_release/cli/commands/publish.py", line [35](https://github.com/.../actions/runs/5754905253/job/15601195357#step:17:36), in publish
    tag = str(tags_and_versions(repo.tags, translator)[0][0])
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
```
2. Running semantic-release version && semantic-release publish silently fails with the following log
```
0.0.1
The next version is: 0.0.1! 🚀
🛠 Running build command: mkdir dist && zip -r dist/coverage.zip ./htmlcov && zip
-r dist/documentation.zip ./docs && pip wheel -w dist .
  adding: htmlcov/ (stored 0%)
  adding: htmlcov/coverage_html.js (deflated 73%)
  adding: htmlcov/d_a44f0ac069e85531___init___py.html (deflated 75%)
  adding: htmlcov/.gitignore (stored 0%)
  adding: htmlcov/status.json (deflated 63%)
  adding: htmlcov/d_7daa3a0111531424___init___py.html (deflated 75%)
  adding: htmlcov/d_7daa3a0111531424_test_jobs_py.html (deflated 79%)
  adding: htmlcov/index.html (deflated 74%)
  adding: htmlcov/favicon_32.png (stored 0%)
  adding: htmlcov/keybd_closed.png (deflated 32%)
  adding: htmlcov/style.css (deflated 79%)
  adding: htmlcov/keybd_open.png (deflated 32%)
  adding: docs/ (stored 0%)
  adding: docs/pdm_db_client/ (stored 0%)
  adding: docs/pdm_db_client/jobs.html (deflated 82%)
  adding: docs/pdm_db_client.html (deflated 68%)
  adding: docs/semantic_release.html (deflated 68%)
  adding: docs/index.html (deflated 68%)
  adding: docs/search.js (deflated 77%)
Processing /home/runner/work/pdm-databricks-client/pdm-databricks-client
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Installing backend dependencies: started
  Installing backend dependencies: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Building wheels for collected packages: pdm-db-client
  Building wheel for pdm-db-client (pyproject.toml): started
  Building wheel for pdm-db-client (pyproject.toml): finished with status 'done'
  Created wheel for pdm-db-client: filename=pdm_db_client-0.0.1-py3-none-any.whl size=3500 sha256=c1d9b9266843d5d1d8030cdaaa3135ea8969d4da8955e2a1e869caf9e91d23d3
  Stored in directory: /home/runner/.cache/pip/wheels/98/9f/d4/1821a5198f53c23e3a86d20867c30244d6829a4249f6ca9168
Successfully built pdm-db-client
[14:37:42] ERROR    [semantic_release.hvcs.github] ERROR           github.py:249
                    github.upload_dists: error uploading asset                  
                    dist/coverage.zip                                           
                    ╭──── Traceback (most recent call last) ─────╮              
                    │ /opt/hostedtoolcache/Python/3.11.4/x64/lib │              
                    │ /python3.11/site-packages/semantic_release │              
                    │ /hvcs/github.py:246 in upload_dists        │              
                    │                                            │              
                    │   243 │   │   │   f for f in glob.glob(dis │              
                    │   244 │   │   ):                           │              
                    │   245 │   │   │   try:                     │              
                    │ ❱ 246 │   │   │   │   self.upload_asset(re │              
                    │   247 │   │   │   │   n_succeeded += 1     │              
                    │   248 │   │   │   except HTTPError:        │              
                    │   249 │   │   │   │   log.error("error upl │              
                    │                                            │              
                    │ /opt/hostedtoolcache/Python/3.11.4/x64/lib │              
                    │ /python3.11/site-packages/semantic_release │              
                    │ /helpers.py:58 in _wrapper                 │              
                    │                                            │              
                    │    55 │   │   │   )                        │              
                    │    56 │   │   │                            │              
                    │    57 │   │   │   # Call function          │              
                    │ ❱  58 │   │   │   result = func(*args, **k │              
                    │    59 │   │   │                            │              
                    │    60 │   │   │   # Log result             │              
                    │    61 │   │   │   logger.debug("%s -> %s", │              
                    │                                            │              
                    │ /opt/hostedtoolcache/Python/3.11.4/x64/lib │              
                    │ /python3.11/site-packages/semantic_release │              
                    │ /hvcs/github.py:208 in upload_asset        │              
                    │                                            │              
                    │   205 │   │   )                            │              
                    │   206 │   │                                │              
                    │   207 │   │   with open(file, "rb") as dat │              
                    │ ❱ 208 │   │   │   response = self.session. │              
                    │   209 │   │   │   │   url,                 │              
                    │   210 │   │   │   │   params={"name": os.p │              
                    │   211 │   │   │   │   headers={            │              
                    │                                            │              
                    │ /opt/hostedtoolcache/Python/3.11.4/x64/lib │              
                    │ /python3.11/site-packages/requests/session │              
                    │ s.py:637 in post                           │              
                    │                                            │              
                    │   634 │   │   :rtype: requests.Response    │              
                    │   635 │   │   """                          │              
                    │   636 │   │                                │              
                    │ ❱ 637 │   │   return self.request("POST",  │              
                    │   638 │                                    │              
                    │   639 │   def put(self, url, data=None, ** │              
                    │   640 │   │   r"""Sends a PUT request. Ret │              
                    │                                            │              
                    │ /opt/hostedtoolcache/Python/3.11.4/x64/lib │              
                    │ /python3.11/site-packages/requests/session │              
                    │ s.py:589 in request                        │              
                    │                                            │              
                    │   586 │   │   │   "allow_redirects": allow │              
                    │   587 │   │   }                            │              
                    │   588 │   │   send_kwargs.update(settings) │              
                    │ ❱ 589 │   │   resp = self.send(prep, **sen │              
                    │   590 │   │                                │              
                    │   591 │   │   return resp                  │              
                    │   592                                      │              
                    │                                            │              
                    │ /opt/hostedtoolcache/Python/3.11.4/x64/lib │              
                    │ /python3.11/site-packages/requests/session │              
                    │ s.py:710 in send                           │              
                    │                                            │              
                    │   707 │   │   r.elapsed = timedelta(second │              
                    │   708 │   │                                │              
                    │   709 │   │   # Response manipulation hook │              
                    │ ❱ 710 │   │   r = dispatch_hook("response" │              
                    │   711 │   │                                │              
                    │   712 │   │   # Persist cookies            │              
                    │   713 │   │   if r.history:                │              
                    │                                            │              
                    │ /opt/hostedtoolcache/Python/3.11.4/x64/lib │              
                    │ /python3.11/site-packages/requests/hooks.p │              
                    │ y:30 in dispatch_hook                      │              
                    │                                            │              
                    │   27 │   │   if hasattr(hooks, "__call__") │              
                    │   28 │   │   │   hooks = [hooks]           │              
                    │   29 │   │   for hook in hooks:            │              
                    │ ❱ 30 │   │   │   _hook_data = hook(hook_da │              
                    │   31 │   │   │   if _hook_data is not None │              
                    │   32 │   │   │   │   hook_data = _hook_dat │              
                    │   33 │   return hook_data                  │              
                    │                                            │              
                    │ /opt/hostedtoolcache/Python/3.11.4/x64/lib │              
                    │ /python3.11/site-packages/semantic_release │              
                    │ /hvcs/util.py:31 in <lambda>               │              
                    │                                            │              
                    │   28 │   """                               │              
                    │   29 │   session = Session()               │              
                    │   30 │   if raise_for_status:              │              
                    │ ❱ 31 │   │   session.hooks = {"response":  │              
                    │   32 │                                     │              
                    │   33 │   if retry:                         │              
                    │   34 │   │   if isinstance(retry, bool):   │              
                    │                                            │              
                    │ /opt/hostedtoolcache/Python/3.11.4/x64/lib │              
                    │ /python3.11/site-packages/requests/models. │              
                    │ py:1021 in raise_for_status                │              
                    │                                            │              
                    │   1018 │   │   │   )                       │              
                    │   1019 │   │                               │              
                    │   1020 │   │   if http_error_msg:          │              
                    │ ❱ 1021 │   │   │   raise HTTPError(http_er │              
                    │   1022 │                                   │              
                    │   1023 │   def close(self):                │              
                    │   1024 │   │   """Releases the connection  │              
                    ╰────────────────────────────────────────────╯              
                    HTTPError: 404 Client Error: Not Found for                  
                    url:                                                        
                    https://api.github.com/repos/.../releases/116079343/as              
                    sets?name=coverage.zip                                      
``` 
3. Running semantic-release publish --tag v0.0.2 silently fails with the following log:
```
[15:49:05] DEBUG    [semantic_release.cli.commands.main] DEBUG       main.py:106
                    main.main: logging level set to: DEBUG                      
           DEBUG    [semantic_release.cli.commands.main] DEBUG       main.py:124
                    main.main: global cli options:                              
                    GlobalCommandLineOptions(noop=False,                        
                    verbosity=2, config_file='pyproject.toml',                  
                    strict=False)                                               
           INFO     [semantic_release.cli.commands.main] INFO        main.py:128
                    main.main: Loading TOML configuration from                  
                    pyproject.toml                                              
           INFO     [semantic_release.cli.config] INFO             config.py:250
                    config.select_branch_options: Using group                   
                    'main' options, as '(main|master)' matches                  
                    'main'                                                      
[[15](https://github.com/.../actions/runs/5834489670/job/15824125389#step:21:16):49:06] DEBUG    [semantic_release.hvcs.util] DEBUG                util.py:45
                    util.build_requests_session: setting up default             
                    session authentication                                      
           DEBUG    [semantic_release.changelog.template] DEBUG   template.py:51
                    template.environment: {'template_dir':                      
                    'templates', 'block_start_string': '{%',                    
                    'block_end_string': '%}',                                   
                    'variable_start_string': '{{',                              
                    'variable_end_string': '}}',                                
                    'comment_start_string': '{#',                               
                    'comment_end_string': '#}',                                 
                    'line_statement_prefix': None,                              
                    'line_comment_prefix': None, 'trim_blocks':                 
                    False, 'lstrip_blocks': False,                              
                    'newline_sequence': '\n',                                   
                    'keep_trailing_newline': False, 'extensions':               
                    (), 'autoescape': True, 'autoescape_value':                 
                    True}                                                       
           DEBUG    [semantic_release.version.translator] DEBUG translator.py:40
                    translator._invert_tag_format_to_re:                        
                    inverted tag_format 'v{version}' to                         
                    'v(?P<version>.*)'                                          
           DEBUG    [semantic_release.cli.masking_filter]   masking_filter.py:30
                    DEBUG masking_filter.add_mask_for:                          
                    Adding redact pattern                                       
                    'context.hvcs_client.token' to                              
                    _redact_patterns                                            
           DEBUG    [semantic_release.cli.masking_filter]   masking_filter.py:30
                    DEBUG masking_filter.add_mask_for:                          
                    Adding redact pattern                                       
                    'context.hvcs_client.token' to                              
                    _redact_patterns                                            
           INFO     [semantic_release.cli.commands.publish] INFO   publish.py:44
                    publish.publish: Uploading distributions to                 
                    release                                                     
           DEBUG    [semantic_release.hvcs.github] DEBUG           helpers.py:47
                    helpers._wrapper:                                           
                    upload_dists(<semantic_release.hvcs.github.Git              
                    hub object at 0x7fe7fdfb2fd0>, tag='v0.0.2',                
                    dist_glob='dist/*')                                         
           DEBUG    [semantic_release.hvcs.github] DEBUG           helpers.py:47
                    helpers._wrapper:                                           
                    get_release_id_by_tag(<semantic_release.hvcs.g              
                    ithub.Github object at 0x7fe7fdfb2fd0>,                     
                    tag='v0.0.2')                                               
           DEBUG    [semantic_release.hvcs.github] DEBUG            github.py:75
                    github._get_repository_owner_and_name: getting              
                    repository owner and name from environment                  
                    variables                                                   
           DEBUG    [urllib3.connectionpool] DEBUG        connectionpool.py:1048
                    connectionpool._new_conn: Starting                          
                    new HTTPS connection (1):                                   
                    api.github.com:443                                          
           DEBUG    [urllib3.connectionpool] DEBUG         connectionpool.py:546
                    connectionpool._make_request:                               
                    https://api.github.com:443 "GET                             
                    /repos/.../releases/tags/v0.0.2                             
                    HTTP/1.1" 200 None                                          
           DEBUG    [semantic_release.hvcs.github] DEBUG           helpers.py:61
                    helpers._wrapper: Github.get_release_id_by_tag              
                    -> 1[16](https://github.com/.../actions/runs/5834489670/job/15824125389#step:21:17)08[37](https://github.com/.../actions/runs/5834489670/job/15824125389#step:21:38)[56](https://github.com/.../actions/runs/5834489670/job/15824125389#step:21:57)                                                
           DEBUG    [semantic_release.hvcs.github] DEBUG           helpers.py:[61](https://github.com/.../actions/runs/5834489670/job/15824125389#step:21:62)
                    helpers._wrapper: Github.upload_dists -> 0
```
4. Running semantic-release publish does nothing i.e. no tags, no artifacts released or published.
I have spent days on this and I've absolutely no clue what to do. Any help would be greatly appreciated.

**Comment by bernardcooke53** [comment]
Hey @eyomi-gilead

Looks like you've had a rough time getting up and running - really appreciate all the logs & detailed description of what's going on, it's very helpful. Let me try and address everything here:

1 - I agree the error isn't particularly helpful, I'm going to get to making it more obvious - what it means is **"looking for the latest version by tags, but couldn't find any tags matching your `tag_format`."** This would be an error condition anyway, even with an improved error, but that's to be expected if you're just getting going on a new project.

2 - This hasn't historically been an error that gets raised, but I'd be open to changing that, particularly in conjunction with `--strict`.

3 - Not too sure here. This looks weird for output:
```
[semantic_release.hvcs.github] DEBUG           helpers.py:61
                    helpers._wrapper: Github.get_release_id_by_tag              
                    -> 1[16](https://github.com/.../actions/runs/5834489670/job/15824125389#step:21:17)08[37](https://github.com/.../actions/runs/5834489670/job/15824125389#step:21:38)[56](https://github.com/.../actions/runs/5834489670/job/15824125389#step:21:57)
```
Is that just formatting from the Github Actions logs? If so it looks like this ran through ok, but `helpers._wrapper: Github.upload_dists -> 0` looks like it couldn't find anything to upload, there would have been error logs otherwise. Is there a chance your working directory's changed (e.g. between CI pipeline steps)?

4 - Publish should just push your wheel/sdist to the release you specify, the tagging/building/release creation should happen within the `version` command.

From your logs it looks like you're trying to do effectively the following (educated guessing):

1. Tag the repo with an appropriate tag
2. Build the project - a coverage report, some documentation and a wheel - all under the `dist/` folder
3. Publish all of the above to a new release created against your Github repo

In which case you should be able to run

```shell
semantic-release version  # will tag the repo, build the project, create a VCS release
semantic-release publish  # will publish your files under `dist/` to the release
```

Just wanted to give a bit of reassurance, I appreciate you've sunk a lot of time into this and I don't think you're missing an obvious trick.

The 404 that comes up when trying to upload assets to Github might be a sign that there's an authentication issue - are you able to use the token you've created to run:

```shell
curl -H "Authorization: bearer $TOKEN" -H "X-GitHub-Api-Version: 2022-11-28" \
    https://api.github.com/repos/<your org>/<your repo>
```

(ripped from [the docs](https://docs.github.com/en/rest/overview/authenticating-to-the-rest-api?apiVersion=2022-11-28) if that helps). That will prove that the token should have permission to read from the repo, which is a start (I had the same issue with private repos in the past). Then it would be a case of checking that you have permission to push to the repo, both locally and in your CI service.

**Comment by bernardcooke53** [comment]
https://github.com/python-semantic-release/python-semantic-release/pull/683 addresses the error message :crossed_fingers:

**Comment by eyomi-gilead** [comment]
@bernardcooke53 I reset everything. I deleted all the tags, all the releases, squashed the commit history.
My Github workflow looks as follows:
```
      - name: Python Semantic Release
        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
        env:
          GH_TOKEN: ${{ secrets.MY_RELEASE_TOKEN }}
        run: |
          git config user.name my-user
          git config user.email my-user@company.com
          semantic-release -vvv version
          semantic-release -vvv publish
```
My pyproject.toml file looks as follows:
```
[tool.semantic_release]
version_variable = "src/semantic_release/__init__.py:__version__"
branch = "main"
upload_to_repository = false
upload_to_release = true
build_command = "mkdir dist && zip -r dist/coverage.zip ./htmlcov && zip -r dist/documentation.zip ./docs && pip wheel -w dist ."
```

**Comment by bernardcooke53** [comment]
Hey @eyomi-gilead 

Glad to hear the tagging is working. I think you need to change:

```toml
version_variable = "src/semantic_release/__init__.py:__version__"
```

to

```toml
version_variables = ["src/semantic_release/__init__.py:__version__"]
```

and this should resolve the issue with the version file not updating.

Another potential cause for the assets not uploading could be fixed by #661 which is just releasing now, so if you can try upgrading to 8.0.7 that would rule out another possibility

**Comment by eyomi-gilead** [comment]
@bernardcooke53 thank you for your suggestions. I was able to get it working after applying the changes you suggested
